csharp/examples/IoTExample/Model/SampleDataset.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Apache.Arrow;
+using Apache.Arrow.Ipc;
+using Apache.Arrow.Memory;
+
+namespace IoTExample
+{
+    public class SampleDataset
+    {
+        private int _size;
+        private readonly int _inputs;
+        private readonly int _capacity;
+        private readonly BlockingCollection<SensorData> _rows;
```

Comment:
Maybe since we are using `async` methods, can we change this to use [Channels](https://devblogs.microsoft.com/dotnet/an-introduction-to-system-threading-channels/) instead? They are very similar, but Channels was made with `async` in mind.

Reply:
Sure, I'll try to use Channel.

Reply:
The code has been refactored, please review.